Library/Homebrew/compat/md5.rb

```diff
@@ -1,26 +1,35 @@
 class Formula
   def self.md5(val)
+    # Don't throw an error at load time, just install time.
     stable.md5(val)
   end
 end
 
 class SoftwareSpec
   def md5(val)
+    odie <<-EOS.undent
+      Using MD5 has been dropped for security reasons.
+      Please switch to SHA-256.
+    EOS
```

Comment:
> I don't think anybody in here is campaigning for keeping MD5 support alive in the sense that we allow formulae that rely on that to be successfully loaded (or even installed).

> We can create a new exception like DeprecatedDSLError . Then in the formula loading we can choose to raise (for developer) or print warning with backtrace (for users).

This sounds a lot like "we should allow users to have a warning but install formulae using `md5`.